src/plugins/plot/vue/single/MctPlot.vue

```diff
@@ -0,0 +1,883 @@
+        },
+
+        startPan(event) {
+            this.canvas.classList.add('plot-drag');
+            this.canvas.classList.remove('plot-marquee');
+
+            this.trackMousePosition(event);
+            this.freeze();
+            this.pan = {
+                start: this.positionOverPlot
+            };
+            event.preventDefault();
+            this.trackHistory();
+
+            return false;
+        },
+
+        updatePan() {
+        // calculate offset between points.  Apply that offset to viewport.
+            if (!this.pan) {
+                return;
+            }
+
+            const dX = this.pan.start.x - this.positionOverPlot.x;
+            const dY = this.pan.start.y - this.positionOverPlot.y;
+            const xRange = this.config.xAxis.get('displayRange');
+            const yRange = this.config.yAxis.get('displayRange');
+
+            this.config.xAxis.set('displayRange', {
+                min: xRange.min + dX,
+                max: xRange.max + dX
+            });
+            this.config.yAxis.set('displayRange', {
+                min: yRange.min + dY,
+                max: yRange.max + dY
+            });
+        },
+
+        trackHistory() {
+            this.plotHistory.push({
+                x: this.config.xAxis.get('displayRange'),
+                y: this.config.yAxis.get('displayRange')
+            });
+        },
+
+        endPan() {
+            this.pan = undefined;
+            this.$emit('userViewportChangeEnd');
+        },
+
+        freeze() {
+            this.config.yAxis.set('frozen', true);
+            this.config.xAxis.set('frozen', true);
+        },
+
+        clear() {
+            this.config.yAxis.set('frozen', false);
+            this.config.xAxis.set('frozen', false);
+            this.plotHistory = [];
+            this.$emit('userViewportChangeEnd');
+        },
+
+        back() {
+            const previousAxisRanges = this.plotHistory.pop();
+            if (this.plotHistory.length === 0) {
+                this.clear();
+
+                return;
+            }
+
+            this.config.xAxis.set('displayRange', previousAxisRanges.x);
+            this.config.yAxis.set('displayRange', previousAxisRanges.y);
+            this.$emit('userViewportChangeEnd');
+        },
+
+        setCursorGuideVisibility(cursorGuide) {
+            this.cursorGuide = cursorGuide === true;
+        },
+
+        setGridLinesVisibility(gridLines) {
+            this.gridLines = gridLines === true;
+        },
+
+        setYAxisKey(yKey) {
+            this.config.series.models[0].emit('change:yKey', yKey);
+        },
+
+        destroy() {
+            configStore.deleteStore(this.config.id);
+
+            this.stopListening();
+            if (this.checkForSize) {
+                clearInterval(this.checkForSize);
+                delete this.checkForSize;
+            }
+
+            if (this.filterObserver) {
+                this.filterObserver();
```

Comment:
Hmm. Adding these to the eventHelper listenTo doesn't quite fit since the helper mostly manages listeners on specific objects but the listening to domain object changes via objects.observe api is a bit different.
I'll think about it again, but right now it doesn't feel right.